src/libexpr/value-to-json.cc

```diff
@@ -61,6 +61,15 @@ json printValueAsJSON(EvalState & state, bool strict,
                     try {
                         out.emplace(state.symbols[a->name], printValueAsJSON(state, strict, *a->value, a->pos, context, copyToStore));
                     } catch (Error & e) {
+
+                        bool isEvalError = dynamic_cast<EvalError *>(&e);
+                        bool isFileNotFoundError = dynamic_cast<FileNotFound *>(&e);
+                        // Restrict replaceEvalErrors only only evaluation errors
+                        if (state.settings.replaceEvalErrors && (isEvalError || isFileNotFoundError)) {
+                            out.emplace(state.symbols[a->name], "«evaluation error»");
```

Comment:
This changes the semantics of `builtins.toJSON` and `derivation` with structured attrs (since it calls `printValueAsJSON()`), making them ignore exceptions which seems like a bad thing.

Reply:
This ties in with the comment about making it a flag instead of a setting.
In line with that, this could then refer to a new C++ parameter on `printValueAsJSON`, instead of `state.settings`.